lib/honeybadger.ex

```diff
@@ -287,10 +288,22 @@ defmodule Honeybadger do
       |> contextual_metadata()
       |> Map.put(:breadcrumbs, breadcrumbs)
 
-    exception
-    |> Notice.new(metadata_with_breadcrumbs, stacktrace, fingerprint)
-    |> put_notice_fingerprint()
-    |> Client.send_notice()
+    notice =
+      exception
+      |> Notice.new(metadata_with_breadcrumbs, stacktrace, fingerprint)
+      |> put_notice_fingerprint()
+
+    exclude_error_value = Application.get_env(:honeybadger, :exclude_errors)
```

Comment:
Per @sorentwo ([link](https://github.com/honeybadger-io/honeybadger-elixir/pull/442#discussion_r936572264)):

> This seems like a compile-time option rather than a runtime option (not to mention this would be a hot path for an application experiencing a rush of errors). What about pulling this up into a compile_env?

Reply:
Fetching the exclude_error using compile_env/2 results to the following error:
```elixir
ERROR! the application :honeybadger has a different value set for key :exclude_errors during runtime compared to compile time. Since this application environment entry was marked as compile time, this difference can lead to different behaviour than expected:

  * Compile time value was not set
  * Runtime value was set to: []

To fix this error, you might:

  * Make the runtime value match the compile time one

  * Recompile your project. If the misconfigured application is a dependency, you may need to run "mix deps.compile honeybadger --force"

  * Alternatively, you can disable this check. If you are using releases, you can set :validate_compile_env to false in your release configuration. If you are using Mix to start your system, you can pass the --no-validate-compile-env flag


** (ErlangError) Erlang error: "aborting bo
```

From the error it suggests we have set a runtime option. Does that mean we also need to set a compile option ? cc @sorentwo

Reply:
It would have to be referenced in the module body, outside of a function. Don't worry about it, there are several places where we're referencing the application environment during runtime. It's just not a recommended practice.